src/pki.rs

```diff
@@ -80,15 +82,15 @@ impl EncryptedKeyblob {
         keyblob.copy_from_slice(&self.0[0x20..]);
 
         let mut cmac = Cmac::<Aes128>::new_varkey(&mac_key.0[..]).unwrap();
-        cmac.input(&self.0[0x10..]);
+        cmac.update(&self.0[0x10..]);
         cmac.verify(&self.0[..0x10])
             .map_err(|err| (keyblob_id, err))?;
 
-        let mut crypter = Ctr128::<Aes128, ZeroPadding>::new_fixkey(
+        let mut crypter = Ctr128::<Aes128>::new(
             GenericArray::from_slice(&key.0),
             GenericArray::from_slice(&self.0[0x10..0x20]),
         );
-        crypter.decrypt_nopad(&mut keyblob)?;
+        crypter.apply_keystream(&mut keyblob);
```

Comment:
Not sure if it's the right thing to do here.

Reply:
Yeah this is correct. Aes CTR is a stream cipher, there never was any padding to begin with!